plugins/snippets/luasnip/default.nix

```diff
@@ -76,7 +90,48 @@ in {
     };
 
     # TODO: add support for snipmate
-    # TODO: add support for lua
+    fromLua = mkOption {
+      default = [];
+      description = ''
+        Load lua snippets with the lua loader.
+        Check https://github.com/L3MON4D3/LuaSnip/blob/master/DOC.md#lua for the necessary file structure.
+      '';
+      example = ''
+        [
+          {}
+          {
+            paths = ./path/to/snippets;
+          }
+        ]
+      '';
+      type = types.listOf (types.submodule {
+        options = {
+          lazyLoad = mkOption {
+            type = types.bool;
+            default = true;
+            description = ''
+              Whether or not to lazy load the snippets
+            '';
+          };
+          paths = mkOption {
+            default = null;
+            type = with types;
+              nullOr (oneOf
+                [
+                  str
+                  path
+                  helpers.rawType
+                  (listOf (oneOf
+                    [
+                      str
+                      path
+                      helpers.rawType
+                    ]))
+                ]);
```

Comment:
to reduce a bit the verbosity I would have done
```nix
let pathTypes = [str path helpers.rawType] in nullOr (oneOf (pathTypes ++ listOf (oneOf pathTypes)))
```

You could even do `pathTypes = oneOf [....]` and use `either` but that would maybe complicate the documentation